src/components/Select/index.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react"
+import {
+  Select as ChakraSelect,
+  SelectProps as ChakraSelectProps,
+  ThemingProps,
+} from "@chakra-ui/react"
+
+interface SelectProps
+  extends Omit<ChakraSelectProps, "defaultValue">,
+    Pick<ThemingProps<"Select">, "variant"> {
+  defaultValue?: { label: string; value: string; [x: string]: any }
+  onChange(selectedOption: any | ""): void
+  placeholder?: string
+  options: Record<string, any>[]
+}
```

Comment:
I was hoping to create a generic type to be able to take the type passed into the `options` array and apply to the payload for `onChange`. But I do not believe I have enough knowledge here to know how to correctly apply it since there are different types of data being sent to the `options` prop based on the four different components that use this component.